src/models/stores/serial.ts

```diff
@@ -0,0 +1,117 @@
+import { NodeChannelInfo } from "src/plugins/dataflow-tool/model/utilities/node";
+
+export class SerialDevice {
+    value: string;
+    localBuffer: string;
+    private port: SerialPort | null;
+    connectChangeStamp: number | null;
+    lastConnectMessage: string | null;
+    deviceInfo: SerialPortInfo | null;
+    serialNodesCount: number;
+    //localPorts: any;
+
+    constructor() {
+      this.value = "0";
+      this.localBuffer = "";
+
+      (navigator as Navigator).serial?.addEventListener("connect", (e) => {
```

Comment:
```suggestion
      navigator.serial?.addEventListener("connect", (e) => {
```
Cast should not be necessary

Reply:
fixed [here](https://github.com/concord-consortium/collaborative-learning/commit/9b1aa6833d63b670554005d766fbcc6a0f34518a)